flang/lib/Lower/OpenMP.cpp

```diff
@@ -2463,10 +2556,113 @@ genTargetOp(Fortran::lower::AbstractConverter &converter,
                  Fortran::parser::OmpClause::Defaultmap>(
       currentLocation, llvm::omp::Directive::OMPD_target);
 
-  return genOpWithBody<mlir::omp::TargetOp>(
-      converter, eval, currentLocation, outerCombined, &clauseList,
-      ifClauseOperand, deviceOperand, threadLimitOperand, nowaitAttr,
-      mapOperands);
+  auto captureImplicitMap = [&](const Fortran::semantics::Symbol &sym) {
```

Comment:
Please add a comment for the piece of code below. I believe it is going through all the symbols in the  construct and creating `MapInfo` for all of them. Please quote the relevant section from the standard that says everything has to be mapped.